deepspeed/runtime/engine.py

```diff
@@ -1091,9 +1129,10 @@ def _configure_optimizer(self, client_optimizer, model_parameters):
                 # If apex/amp is available it will be imported above
                 raise RuntimeError(
                     "Unable to import apex/amp, please make sure it is installed")
-            self.module, self.optimizer = amp.initialize(
+            model, self.optimizer = amp.initialize(
```

Comment:
Does self.module also need updating here?

Reply:
It gets set via `_set_client_model(model)` on the next line, we have to be careful how to assign `self.module` now so i factored it out into this helper method.